### PR TITLE
In memory cache and misc fixes

### PR DESCRIPTION
**Pull request by elia**
Hey @janbiedermann I was finally able to properly try out OWL! 🎉

Let me know if the cache I introduced makes sense for webpacker and if I missed some use of `.owl_cache` (I didn't found any).

### CODE REVIEW

**Review by janbiedermann — COMMENTED** [review]
That doesn't work, because Oj will make that in json ":key":"value", vs "key":"value".
I am using Oj with default options for maximum speed.

**Comment by janbiedermann** [comment]
Thanks a lot @elia!
I cherry-picked all except c408275 and 1e783bd.
I will experiment with the cache a bit in the next couple of days. For 1e783bd see comment above.
I leave this PR open, until i came to a conclusion regarding the cache.

File: index.js [hunk 1]
```diff
@@ -160,21 +171,17 @@ function start_compile_server() {
...
-    if (typeof Owl.options.hmrHook === 'undefined') { Owl.options.hmrHook = ''; }
-    if (typeof Owl.options.sourceMap === 'undefined') { Owl.options.sourceMap = false; }
-    if (typeof Owl.options.includePaths === 'undefined') { Owl.options.includePaths = null; }
-    if (typeof Owl.options.requireModules === 'undefined') { Owl.options.requireModules = null; }
-    if (typeof Owl.options.dynamicRequireSeverity === 'undefined') { Owl.options.dynamicRequireSeverity = null; }
-    if (typeof Owl.options.compilerFlagsOn === 'undefined') { Owl.options.compilerFlagsOn = null; }
-    if (typeof Owl.options.compilerFlagsOff === 'undefined') { Owl.options.compilerFlagsOff = null; }
+    const options = loaderUtils.getOptions(that);
+    Object.keys(default_options).forEach(
+        (key) => if (option[key] === undefined) option[key] = default_options[key];
```
Comment (janbiedermann) [inline index.js]: must be options[key] and the whole thing after => in curly brackets, i fixed it.

**Comment by elia** [comment]
Thanks for the detailed explanation, I'm still learning all the intricacies of webpack, glad you cherry-picked the other commits 👍

**Comment by janbiedermann** [comment]
> I am using Oj with default options for maximum speed.
Thats actually wrong, mode: :strict seems to be faster (and safer). Changed it already in master.

**Comment by janbiedermann** [comment]
With one project where multiple assets are build, compilations now take over 1 minute, especially when containerized. A shared cache, maybe via memcached, becomes very attractive.
In a container with such setup can be easily shipped.

**Pull request closed without merging**